tests/unit/customizations/cloudformation/test_yamlhelper.py

```diff
@@ -94,6 +103,70 @@ def test_parse_json_with_tabs(self):
         output = yaml_parse(template)
         self.assertEqual(output, {'foo': 'bar'})
 
+    def test_parse_json_preserve_elements_order(self):
+        input_template = """
+        {
+            "B_Resource": {
+                "Key2": {
+                    "Name": "name2"
+                },
+                "Key1": {
+                    "Name": "name1"
+                }
+            },
+            "A_Resource": {
+                "Key2": {
+                    "Name": "name2"
+                },
+                "Key1": {
+                    "Name": "name1"
+                }
+            }
+        }
+        """
+        expected_dict = OrderedDict([
+            ('B_Resource', OrderedDict([('Key2', {'Name': 'name2'}), ('Key1', {'Name': 'name1'})])),
+            ('A_Resource', OrderedDict([('Key2', {'Name': 'name2'}), ('Key1', {'Name': 'name1'})]))
+        ])
+        output_dict = yaml_parse(input_template)
+        self.assertEqual(expected_dict, output_dict)
+
+    def test_parse_yaml_preserve_elements_order(self):
+        input_template = """
+        B_Resource:
+            Key2:
+                Name: name2
+            Key1:
+                Name: name1
+        A_Resource:
+            Key2:
+                Name: name2
+            Key1:
+                Name: name1
+        """
+        output_dict = yaml_parse(input_template)
+        expected_dict = OrderedDict([
+            ('B_Resource', OrderedDict([('Key2', {'Name': 'name2'}), ('Key1', {'Name': 'name1'})])),
+            ('A_Resource', OrderedDict([('Key2', {'Name': 'name2'}), ('Key1', {'Name': 'name1'})]))
+        ])
+        self.assertEqual(expected_dict, output_dict)
+
+        output_template = yaml_dump(output_dict)
+        # yaml dump changes indentation, remove spaces and new line characters to just compare the text
```

Comment:
Is there a reason to do this? Seems overly complex when you could just do an equality check. The `input_template` can just be made to have the same indentation as the expected output no?

```python
 input_template = (
    'B_Resource:\n'
    '  Key2:\n'
    '    Name: name2\n'
    '  Key1:\n'
    '    Name: name1\n'
    'A_Resource:\n'
    '  Key2:\n'
    '    Name: name2\n'
    '  Key1:\n'
    '    Name: name1\n'
)
```